### PR TITLE
Fix sockpool wait_for_fd for old releases

### PR DESCRIPTION
**Pull request by Coldwings**
Same as #553 but for dev/0.6

### CODE REVIEW

File: io/epoll.cpp [hunk 1]
```diff
@@ -266,22 +266,38 @@ class EventEngineEPoll : public MasterEventEngine, public CascadingEventEngine {
...
-        } else if (ret == 0) {
-            rm_interest(event);  // Timeout
-            errno = ETIMEDOUT;
-            return -1;
-        } else {
-            rm_interest(event);  // Interrupted by other thread
-            errno = err.no;
-            return -1;
         }
+        rm_interest({fd, interests, 0}); // no ONE_SHOT, to reconfig epoll
```
Comment (lihuiba) [inline io/epoll.cpp:297]: it's might be safer to ```rm_interest(event)``` as before.

File: io/kqueue.cpp [hunk 2]
```diff
@@ -164,11 +187,12 @@ class KQueue : public MasterEventEngine, public CascadingEventEngine {
...
 
     ssize_t wait_for_events(void** data,
-            size_t count, uint64_t timeout = -1) override {
-        int ret = get_vcpu()->master_event_engine->wait_for_fd_readable(_kq, timeout);
+            size_t count, uint64_t timeout) override {
+        int ret = ::photon::wait_for_fd_readable(_kq, timeout);
         if (ret < 0) return errno == ETIMEDOUT ? 0 : -1;
         if (count > LEN(_events))
             count = LEN(_events);
+        static const struct timespec _tm = {0, 0};
```
Comment (lihuiba) [inline io/kqueue.cpp:195]: Why static here, while not in enqueue()?

File: io/epoll.cpp (hunk 1 above)
Comment (Coldwings) [reply io/epoll.cpp]: done

File: io/kqueue.cpp (hunk 2 above)
Comment (Coldwings) [reply io/kqueue.cpp]: Use on-stack temporary timespec is fine. Changed in latest commit.

**Comment by Coldwings** [comment]
Discussed that old version may not need such fix. This fix will be dropped.

**Pull request closed without merging**